### DISCUSSION TITLE
Are subscription triggers scoped to the current schema?

### DISCUSSION THREAD

**Discussion by janko** [rmosolgo/graphql-ruby#5079] [General]
As part of integration testing subscription triggers with Action Cable, I noticed the internal stream name that was generated didn't appear to embed the current schema identifier. I went digging in `graphql` gem code, but I couldn't find concrete proof that a triggered subscription update won't be broadcasted across *all* schemas.

We have multiple GQL schemas, and one are v1 & v2 versions of the same API, which share many of the same subscription fields. For example, let's say I have the following:

```rb
class MyApi::V1
  class SubscriptionType < GraphQL::Schema::Object
    field :some_event, ...
  end

  class Schema < GraphQL::Schema
    use GraphQL::Subscriptions::ActionCableSubscriptions
    subscription(SubscriptionType)
  end
end
```
```rb
class MyApi::V2
  class SubscriptionType < GraphQL::Schema::Object
    field :some_event, ...
  end

  class Schema < GraphQL::Schema
    use GraphQL::Subscriptions::ActionCableSubscriptions
    subscription(SubscriptionType)
  end
end
```

Then one subscriber subscribes to v1 schema's `someEvent { ... }` and another subscriber subscribes to v2 schema's `someEvent { ... }`. If I call `MyApi::V1.subscriptions.trigger(:some_event, {}, {})`, what part of the `graphql` gem will ensure that only the first subscriber's (subscribed to v1 schema) subscription will get updated and not the second one (subscribed to v2 schema)?

**Comment by rmosolgo** [comment]
Hey, great question. With ActionCable, you can add a `namespace:`, for example: 

```ruby 
use GraphQL::Subscriptions::ActionCableSubscriptions, namespace: "V2" # or "V1", etc
```

It will be included in the stream identifiers later on: 

https://github.com/rmosolgo/graphql-ruby/blob/4836730486ea77714a8343ad5805cd4cdff0dc3e/lib/graphql/subscriptions/action_cable_subscriptions.rb#L245-L251


Does that work for you?

**Reply by janko** [reply]
Yes, that works for us 👍🏻 I discovered this when I was reading the source code, I didn't see it mentioned in the guides, so I wasn't sure whether it was intended for this use case. Glad that there is any easy solution 🙂
